File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/service/PlzService.java
package at.ac.tuwien.sepm.groupphase.backend.service;

import at.ac.tuwien.sepm.groupphase.backend.dto.PlzDto;

import java.util.List;

public interface PlzService {

    /**
     * Finds the top 10 plzs that start with the given string.
     *
     * @param plzQuery plz string to search for
     * @return found plzs
     */
    List<PlzDto> findPlzSuggestions(String plzQuery);

}

File: frontend/src/app/services/plz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Globals } from 'src/app/global';
import { Plz } from 'src/app/dtos';

@Injectable({
  providedIn: 'root',
})
export class PlzService {
  private plzBaseUri: string = this.globals.backendUri + '/plz';

  constructor(private httpClient: HttpClient, private globals: Globals) {}

  /**
   * Finds the top 10 plzs that start with the given string.
   *
   * @param plz plz string to search for
   * @returns found plzs
   */
  findSuggestions(plz: string): Observable<Plz[]> {
    console.log(`Search for plz suggestions with plz=${plz}`);
    return this.httpClient.get<Plz[]>(`${this.plzBaseUri}`, {
      params: { plz },
    });
  }
}
